src/Ninject.Web.AspNetCore/NinjectServiceProviderFactory.cs

```diff
@@ -8,13 +8,17 @@ public class NinjectServiceProviderFactory : IServiceProviderFactory<NinjectServ
 		private readonly IKernel _kernel;
 
 
-		public NinjectServiceProviderFactory() : this(new StandardKernel())
+		public NinjectServiceProviderFactory() : this(new AspNetCoreKernel())
 		{
 		}
 
 		public NinjectServiceProviderFactory(IKernel kernel)
```

Comment:
Instead of having here IKernel as argument type, we could use instead AspNetCoreKernel? Then we would not need the check there and it's statically ensured that we pass a AspNetCoreKernel?

Reply:
Ah but it doesn't help as the NinjectSelfHostBootstrapper accepts an IKernel and there we could pass the wrong one.